examples/UnreleasedAI/UnreleasedAI.md

```diff
@@ -0,0 +1,163 @@
+
+To create a prompt-based scorer, you define a prompt that classifies its arguments, and a scoring function that converts the classification choices into scores. In [eval/comprehensiveness-scorer.ts](https://github.com/braintrustdata/unreleased-ai/blob/6e74be5caed1a1c368ee7124a5adc7e0c27f2969/eval/comprehensiveness-scorer.ts#L9), we defined our prompt as:
+
+```typescript
+const promptTemplate = `You are an expert technical writer who helps assess how effectively an open source product team generates a changelog based on git commits since the last release. Analyze commit messages and determine if the changelog is comprehensive, accurate, and informative. 
+
+Input:  
+{{input}}  
+
+Changelog:  
+{{output}}  
+
+Assess the comprehensiveness of the changelog and select one of the following options. List out which commits are missing from the changelog if it is not comprehensive.  
+a) The changelog is comprehensive and includes all relevant commits  
+b) The changelog is mostly comprehensive but is missing a few commits  
+c) The changelog includes changes that are not in commit messages  
+d) The changelog is incomplete and not informative`;
+```
+
+Writing a prompt to use for these types of evaluations is difficult. In fact, it may take many iterations to come up with a prompt that you believe judges the output correctly. To refine this iteration process, you can even upload this prompt to Braintrust and call it as a function.
+
+## Evals
+
+Now, let’s use the comprehensiveness scorer to create a feedback loop that allows us to iterate on our prompt and make sure we’re shipping a reliable, high quality product. In Braintrust, you can run evaluations, or [Evals](https://www.braintrust.dev/docs/guides/evals/run), if you have a Task, Scores, and Dataset. We have a task, which is the invoke function we’re calling in our app. We have scores, the comprehensiveness function we just defined to assess the quality of our function outputs. The final piece we need to run evaluations is a [dataset](https://www.braintrust.dev/docs/guides/datasets).
+
+### Datasets
+
+Go to your Braintrust **Logs** and select the logs with undesirable outputs – maybe your input was invalid or not found, or maybe the output just wasn’t good enough. Use these logs to create a dataset called eval dataset, which we’ll evaluate for comprehensiveness to understand where the prompt might need adjustments. 
+
+![Adding error logs to dataset](assets/add-logs-to-dataset.png)
+
+Alternatively, you can define a dataset in [eval/sampleData.ts](https://github.com/braintrustdata/unreleased-ai/blob/main/eval/sampleData.ts).
+
+Now that we have all three inputs, we can establish an Eval() function in [eval/changelog.eval.ts](https://github.com/braintrustdata/unreleased-ai/blob/6e74be5caed1a1c368ee7124a5adc7e0c27f2969/eval/changelog.eval.ts#L26C1-L36C4):
+
+```typescript
+Eval(PROJECT_NAME, {  
+    data: () => [sampleData],  
```

Comment:
should we change this to `initDataset` to reference the newly created dataset? You could do something like:

```
data: initDataset({project: PROJET_NAME, dataset: "My dataset name"}),
//  data: () => [sampleData], // Uncomment to use sample data
```

Reply:
yeah let me swap it in the code repo